docs/001_develop/03_client-capabilities/023_custom-components/03_state-management

```diff
@@ -0,0 +1,342 @@
+    }
+}
+```
+
+The `label` property is tied to an HTML attribute, so to override the value of `label` we can simply replace it in the HTML.
+
+The default value of `label` is 'Submit', if no value is provided in the HTML.
+
+```shell
+ <my-button label="Cancel"></my-buttonn>
+```
+
+:::note
+
+The `labelChanged` method is only useful if you want to perform additional actions when the label attribute changes, such as logging or triggering other side effects. Since the `@attr` decorator already handles the synchronization between the attribute and property (updating the DOM whenever label changes), the labelChanged method isn’t strictly necessary.
+
+:::
+
+:::note
+Here's a list of attribute characteristics:
+
+- Always strings in HTML
+- Used in HTML markup
+- Visible in DOM inspector
+- Used for initial state
+- Slower for dynamic updates
+  :::
+
+### Attribute Value Binding
+
+For most attributes, we can directly bind their values to expressions in the template, allowing them to update dynamically based on properties in the component.
+
+The label attribute controls the text shown on the button. To bind label to a dynamic value, we use an expression within the template. Here’s how this might look in a <my-button> component:
+
+```shell
+import { attr, customElement, GenesisElement, html } from '@genesislcap/web-core';
+
+const template = html<MyButton>`
+    <button>${x => x.label}</button>
+`;
+
+@customElement({
+  name: 'my-button',
+  template,
+  styles,
+})
+
+export class MyButton extends GenesisElement {
+    @attr label = 'Click Me';
+    # Here we are binding the text inside the button to the `label` attribute.
+});
+```
+
+#### Usage in HTML
+
+```shell
+<my-button label="Cancel"></my-button>
+```
+
+- Here, the label attribute binds directly to the button's text via `${x => x.label}`.
+- If label is set to "Cancel" in HTML `(<my-button label="Cancel">)`, the button will display "Cancel".
+- When the label value is changed dynamically, the button text will automatically update.-
+
+### Interpolated Attribute Values
+
+In cases where an attribute value is partially static and partially dynamic, we can combine static and dynamic parts using interpolated expressions. This approach is useful for creating customized button labels or CSS classes.
+
+Suppose we want to add a counter to the button label that shows the number of times it has been clicked. We could use an expression to combine the static text ("Clicked: "), and a dynamic value, such as the clickCount property:
+
+```shell
+import { attr, customElement, GenesisElement, html } from '@genesislcap/web-core';
+
+const template = html<MyButton>`
+    <button>${x => `${x.label} (Clicked: ${x.clickCount} times)`}</button>
+`;
+
+@customElement({
+  name: 'my-button',
+  template,
+  styles,
+})
+
+export class MyButton extends FASTElement {
+    @attr label = 'Click Me'; // Bind to 'label' attribute
+    clickCount = 0;
```

Comment:
I think this will need the `@observable` decorator or the template won't update

Reply:
nicely spotted: I actually need to showcase interpolated attribute values here so I'll be updating example so it doesn't need reactivity